api/ws/workers/middlewares/emit.js

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const connectionsTable = require('../../workers/connections_table');
+
+/**
+ * @param {object} request
+ */
+function addNonceToRequest(request) {
+	request.data.nonce = connectionsTable.getNonce(request.socket.id);
```

Comment:
We should check if `request.data` exists and is not null before assigning a nonce property to it (else it could crash/reboot the worker) - A rogue/malicious client could send null as the data if they do `socket.emit('postBlock', null)`.

Reply:
Will do, thanks!